typescript/types/locale.d.ts

```diff
@@ -3,10 +3,12 @@ declare module Locale {
 
   type Locale = {
     name: string,
+    /** @deprecated */
```

Comment:
Let's add a message and link to this MR and #3269 so people can see the rationale too (and perhaps for the console warning above too)

Reply:
Sure, will do later